monitor/services/src/main/java/org/datacleaner/monitor/server/filesystem/AbstractWaitForCompleteFileStrategy.java

```diff
@@ -0,0 +1,48 @@
+/**
+ * DataCleaner (community edition)
+ * Copyright (C) 2014 Neopost - Customer Information Management
+ *
+ * This copyrighted material is made available to anyone wishing to use, modify,
+ * copy, or redistribute it subject to the terms and conditions of the GNU
+ * Lesser General Public License, as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this distribution; if not, write to:
+ * Free Software Foundation, Inc.
+ * 51 Franklin Street, Fifth Floor
+ * Boston, MA  02110-1301  USA
+ */
+package org.datacleaner.monitor.server.filesystem;
+
+import java.io.File;
+
+public abstract class AbstractWaitForCompleteFileStrategy implements WaitForCompleteFileStrategy {
+    protected static final int ATTEMPT_WAIT_INTERVAL_MS = 500;
+    protected static final int WAIT_ATTEMPT_LIMIT = 20 * 60 * (2 * ATTEMPT_WAIT_INTERVAL_MS);
```

Comment:
These should probably be configurable as well. For example the `GeneralWaitForCompleteFileStrategy` might need a little (or a lot, actually) more time than half a second, depending on the program, chunk sizes and quality of Internet connection when downloading.

We might want to specify the total limit as time instead of attempts. Attempts make sense when retrying something that failed, but a timeout probably makes more sense in this case.